src/main/java/com/vesta/controller/SubjectController.java

```diff
@@ -33,4 +35,22 @@
     @ApiOperation(value = "Returns the all subjects")
     @GetMapping("/subjects/all")
     List<SubjectView> getAll();
+
+    @ApiOperation(value = "Returns the all subjects by FloorID")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Get subjects by FlooID has succeeded"),
+            @ApiResponse(code = 403, message = "Accessing the resource you were trying to reach is forbidden"),
+            @ApiResponse(code = 404, message = "Floor not found")
+    })
+    @GetMapping("/subjects/all/by/floor")
+    List<SubjectView> getAllByFloorId(Long floorId);
+
+    @ApiOperation(value = "Add a new subject")
+    @ApiResponses(value = {
+            @ApiResponse(code = 201, message = "Subject was added"),
+            @ApiResponse(code = 500, message = "Server error")
+    })
+    @PostMapping("/subjects/add/new/subject")
```

Comment:
Change the path "/subjects/add/new/subject" to "/subjects", this is a post mapping which mean that you want to creat something so where is no need to specifie "add/new/subject"